apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -81,50 +77,41 @@ public void ExceptionAfterRetryExhaustedTest()
 
         AssertAttempts(
            retryOptions: retryOptions,
-           statusCode: 502,
-           numOfFailures: 4,
-           maximumRetries: 3,
+           responseStatusCodes: new[] { 502, 502, 502 },
            success: false,
            expectedBackOffs: expectedBackOffs);
     }
 
-    private static void AssertAttempts(RetryOptions retryOptions, int statusCode, int numOfFailures, int maximumRetries, bool success, params int[] expectedBackOffs)
+    private static void AssertAttempts(RetryOptions retryOptions, int[] responseStatusCodes, bool success, params int[] expectedBackOffs)
```

Comment:
This method would just accept an instance of the new test class that specifies everything that's involved.

There can then be one TheoryData for timing tests, and another for predicate tests - or we could just put everything together, to be honest.

See https://github.com/googleapis/gax-dotnet/blob/daa2e0b8591ac2a3e995f822b526315482ddd832/Google.Api.Gax.Grpc.IntegrationTests/ClientBuilderBaseTest.cs#L245 for an example of code that creates a TheoryData for a custom class.